javalib/src/main/scala/java/util/Collections.scala

```diff
@@ -0,0 +1,1194 @@
+
+  def newSetFromMap[E](map: Map[E, Boolean]): Set[E] = {
+    if (map.nonEmpty)
+      throw new IllegalArgumentException
+
+    new WrappedSet[E, Set[E]] {
+      override protected val inner: Set[E] =
+        map.keySet
+
+      override def add(e: E): Boolean =
+        map.put(e, true) === null
+
+      override def addAll(c: Collection[_ <: E]): Boolean =
+        c.foldLeft(false)((prev, elem) => map.put(elem, true) === null || prev)
+    }
+  }
+
+  @inline
+  private def modulo(a: Int, b: Int): Int = ((a % b) + b) % b
+
+  @inline
+  private def naturalComparator[T]: Comparator[T] = {
+    new Comparator[T with jl.Comparable[T]] {
+      final def compare(o1: T with jl.Comparable[T], o2: T with jl.Comparable[T]): Int = o1.compareTo(o2)
+    }.asInstanceOf[Comparator[T]]
+  }
+
+  @inline
+  private implicit def comparatorToOrdering[E](cmp: Comparator[E]): Ordering[E] = {
+    new Ordering[E] {
+      final def compare(x: E, y: E): Int = cmp.compare(x, y)
+    }
+  }
+
+  @inline
+  private def binarySearchImpl[E](list: List[E], compareToKey: E => Int): Int = {
+    def notFound(insertionPoint: Int): Int = {
+      -insertionPoint - 1
+    }
+
+    @tailrec
+    def binarySearch(lo: Int, hi: Int, get: Int => E): Int = {
+      if (lo < hi) {
+        val mid = lo + (hi - lo) / 2
+        val cmp = compareToKey(get(mid))
+        if (cmp == 0) mid
+        else if (cmp > 0) binarySearch(lo, mid, get)
+        else binarySearch(mid + 1, hi, get)
+      } else {
+        notFound(lo)
+      }
+    }
+
+    list match {
+      case _: RandomAccess =>
+        binarySearch(0, list.size, list(_))
+
+      case _ =>
+        def getFrom(iter: ListIterator[E])(index: Int): E = {
+          val shift = index - iter.nextIndex
+          if (shift > 0) (0 until shift).foreach(_ => iter.next())
+          else (0 until -shift).foreach(_ => iter.previous())
+          iter.next()
+        }
+        binarySearch(0, list.size, getFrom(list.listIterator))
+    }
+  }
+
+  @inline
+  def reverseImpl[T](list: List[T]): Unit = {
+    val size = list.size
+    list match {
+      case list: RandomAccess =>
+        for (i <- 0 until size / 2) {
+          val tmp = list(i)
+          list(i) = list(size - i - 1)
+          list(size - i - 1) = tmp
+        }
+
+      case _ =>
+        val it1 = list.listIterator()
+        val it2 = list.listIterator(size)
+        for (i <- 0 until size / 2) {
+          val tmp = it1.next()
+          it1.set(it2.previous())
+          it2.set(tmp)
+        }
+    }
+  }
+
+  @inline
+  private def indexOfSubListImpl(source: List[_], target: List[_], fromStart: Boolean): Int = {
+    val targetSize = target.size
+    if (targetSize == 0) {
+      0
+    } else {
+      val indices = 0 to source.size - targetSize
+      val indicesInOrder = if (fromStart) indices else indices.reverse
+      indicesInOrder.find { i => source.subList(i, i + target.size).equals(target) }.getOrElse(-1)
```

Comment:
Put the body of the brace-lambda on its own line.
It will also make this line shorter.